src/batches/setBatchResults.m

```diff
@@ -28,53 +22,77 @@
+  matlabbatch{end}.spm.stats.results.conspec.titlestr = returnName(result);
 
+  matlabbatch{end}.spm.stats.results.conspec.contrasts = result.contrastNb;
+  matlabbatch{end}.spm.stats.results.conspec.threshdesc = result.Contrasts.MC;
+  matlabbatch{end}.spm.stats.results.conspec.thresh = result.Contrasts.p;
+  matlabbatch{end}.spm.stats.results.conspec.extent = result.Contrasts.k;
   matlabbatch{end}.spm.stats.results.conspec.conjunction = 1;
-
-  matlabbatch{end}.spm.stats.results.conspec.mask.none = ...
-      ~opt.result.Steps(iStep).Contrasts(iCon).Mask;
+  matlabbatch{end}.spm.stats.results.conspec.mask.none = ~result.Contrasts.useMask;
 
   matlabbatch{end}.spm.stats.results.units = 1;
 
-  % TODO
-  % add flags to make those optional
-  matlabbatch{end}.spm.stats.results.export{1}.png = true;
-  matlabbatch{end}.spm.stats.results.export{2}.csv = true;
-  matlabbatch{end}.spm.stats.results.export{3}.tspm.basename = ...
-      opt.result.Steps(iStep).Contrasts(iCon).Name;
+  matlabbatch{end}.spm.stats.results.export = [];
+  if result.Output.png
+    matlabbatch{end}.spm.stats.results.export{end + 1}.png = true;
+  end
 
-  % TODO
-  % add the possibility to create a montage
-  %   matlabbatch{1}.spm.stats.results.export{3}.montage.background = '<UNDEFINED>';
-  %   matlabbatch{1}.spm.stats.results.export{3}.montage.orientation = '<UNDEFINED>';
-  %   matlabbatch{1}.spm.stats.results.export{3}.montage.slices = '<UNDEFINED>';
+  if result.Output.csv
+    matlabbatch{end}.spm.stats.results.export{end + 1}.csv = true;
+  end
 
-  if opt.result.Steps(1).Contrasts(iCon).NIDM
+  if result.Output.thresh_spm
+    matlabbatch{end}.spm.stats.results.export{end + 1}.tspm.basename = returnName(result);
+  end
+
+  if result.Output.binary
+    matlabbatch{end}.spm.stats.results.export{end + 1}.binary.basename = [returnName(result), ...
+                                                                          '_mask'];
+  end
+
+  if result.Output.NIDM_results
 
     matlabbatch{end}.spm.stats.results.export{end + 1}.nidm.modality = 'FMRI';
 
     matlabbatch{end}.spm.stats.results.export{end}.nidm.refspace = 'ixi';
-    if strcmp(opt.space, 'T1w')
+    if strcmp(result.space, 'individual')
       matlabbatch{end}.spm.stats.results.export{end}.nidm.refspace = 'subject';
     end
 
-    matlabbatch{end}.spm.stats.results.export{end}.nidm.group.nsubj = results.nbSubj;
+    matlabbatch{end}.spm.stats.results.export{end}.nidm.group.nsubj = result.nbSubj;
+
+    matlabbatch{end}.spm.stats.results.export{end}.nidm.group.label = result.label;
 
-    matlabbatch{end}.spm.stats.results.export{end}.nidm.group.label = results.label;
+  end
+
+  if result.Output.montage.do
+
+    matlabbatch{end}.spm.stats.results.export{end + 1}.montage = setMontage(result);
+
+    % Not sure why the name of the figure does not come out right
+    matlabbatch{end + 1}.spm.util.print.fname = ['Montage' returnName(result)];
```

Comment:
@Remi-Gau not sure if this one is working on mac computer, maybe @marcobarilari has tried already? because in mine, `returnName(result)` output is for instance `'AllCateg_p-0.050_k-0_MC-FWE'`, it does save the file under: `MontageAllCateg_p-0_001.png` label

Reply:
If you are OK with this output: `MontageAllCateg_p-0050_k-0_MC-FWE_001.png` I can make a PR with either the below in `returnName` function:

```
  name = sprintf('%s_p-%s_k-%i_MC-%s', ...
                 result.Contrasts.Name, ...
                 erase(sprintf('%0.3f',result.Contrasts.p),'.'), ...
                 result.Contrasts.k, ...
                 result.Contrasts.MC);
```

if erase function is not octave compatible, below could be used:

`num2str(100*result.Contrasts.p,'%03d')`

But you could easily say p-0050 is not super transparent to understand.

Reply:
Actually you can use `strrep` (string replace) and it will work on octave.

```
  name = sprintf('%s_p-%s_k-%i_MC-%s', ...
                 result.Contrasts.Name, ...
                 erase(sprintf('%0.3f',result.Contrasts.p),'.'), ...
                 result.Contrasts.k, ...
                 result.Contrasts.MC);

  name = strrep(name, '.', '');
```

Reply:
I think I had noticed when coding this but then forgotten.

I did not understand at the time but it seems that spm saving figures just takes some liberty to say that the first `.` in a filename must denote the extension of the file. Not a bad heuristic in most cases.

Reply:
Feel free to open a PR

:-)